### PR TITLE
Fixes #32452 - Host Registration tweaks and improvements

### PR DESCRIPTION
**Pull request by stejskalleos**
- Updated ApiPie documentation
- Fixed typo in `default_host_init_config_template` setting
- Snippet `host_init_config_post` moved at the end of the `host_init_config`
  template, where it should be, not before `insights` snippet
- Use `title` instead of `name` for host groups and operating systems
- Updated list of variables in `global_registration` template

### CODE REVIEW

File: app/models/setting/provisioning.rb [hunk 1]
```diff
@@ -88,7 +88,7 @@ def self.default_settings
       ),
       set('maximum_structured_facts', N_("Maximum amount of keys in structured subtree, statistics stored in foreman::dropped_subtree_facts"), 100, N_('Maximum structured facts')),
       set('default_global_registration_item', N_("Global Registration template"), 'Global Registration', N_("Default Global registration template")),
-      set('default_host_init_config_template', N_("Default 'Host intial configuration' template, automatically assigned when new operating system is created"), 'Linux host_init_config default', N_("Default 'Host intial configuration' template")),
+      set('default_host_init_config_template', N_("Default 'Host initial configuration' template, automatically assigned when new operating system is created"), 'Linux host_init_config default', N_("Default 'Host initial configuration' template")),
```
Comment (ekohl) [inline app/models/setting/provisioning.rb:91]: This typo fix feels like something that should be cherry picked, at least to 2.5-stable.

File: app/views/unattended/provisioning_templates/registration/global_registration.erb [hunk 2]
```diff
@@ -13,12 +13,17 @@ model: ProvisioningTemplate
 <%= "# User: [#{@user.login}]" -%>
 <%= "\n# Organization: [#{@organization.name}]" if @organization -%>
 <%= "\n# Location: [#{@location.name}]" if @location -%>
-<%= "\n# Hostgroup: [#{@hostgroup.name}]" if @hostgroup -%>
-<%= "\n# Operating System: [#{@operatingsystem.name}]" if @operatingsystem -%>
-<%= "\n# Setup Insights: [#{@setup_insights}]" unless @setup_insights.nil? -%>
-<%= "\n# Setup Remote Execution: [#{@setup_remote_execution}]" unless @setup_remote_execution.nil? -%>
+<%= "\n# Host group: [#{@hostgroup.title}]" if @hostgroup -%>
+<%= "\n# Operating system: [#{@operatingsystem}]" if @operatingsystem -%>
+<%= "\n# Setup insights: [#{@setup_insights}]" unless @setup_insights.nil? -%>
```
Comment (ekohl) [inline app/views/unattended/provisioning_templates/registration/global_registration.erb]: I think this shouldn't changed. Insights is a brand name and deserves a capital letter.

**Comment by ekohl** [comment]
Looking at the patch I think it may be safe enough to cherry pick entirely into 2.5.

**Comment by stejskalleos** [comment]
@ekohl Updated

> Looking at the patch I think it may be safe enough to cherry pick entirely into 2.5.

I believe so, the changes are more or less just visual stuff, except the `title` for host group and OS but that is small change.

**Review by ekohl — DISMISSED** [review]
@ezr-ondrej I see you assigned this to yourself so I'm not pressing the merge button, but LGTM.

**Comment by stejskalleos** [comment]
@ezr-ondrej Can we merge the PR? Would be nice to get it to 2.5 as soon as possible.

Javascript tests failures are not related to the changes in the PR:
```
/home/runner/work/foreman/foreman/webpack/assets/javascripts/react_app/Root/apollo.js
Error:   2:1  error  '@apollo/client/link/batch-http' should be listed in the project's dependencies. Run 'npm i -S @apollo/client/link/batch-http' to add it  import/no-extraneous-dependencies
```

I didn't touch that code at all.

**Review by ares — APPROVED** [review]
The errors are unrelated, I saw them in other PRs too. I tested the functionality, nice tweaks, thanks @stejskalleos, merging!

**Comment by ares** [comment]
also this should go to 2.5-stable most likely

**Pull request merged**